marshmallow/schema.py

```diff
@@ -583,13 +586,17 @@ def _do_load(self, data, many=None, partial=None, postprocess=True):
         :param data: The data to deserialize.
         :param bool many: Whether to deserialize `data` as a collection. If `None`, the
             value for `self.many` is used.
-        :param bool partial: Whether to ignore missing fields. If `None`, the
-            value for `self.partial` is used.
+        :param bool|tuple partial: Whether to ignore missing fields. If `None`,
+            the value for `self.partial` is used. If its value is an iterable,
+            only missing fields listed in that iterable will be ignored.
         :param bool postprocess: Whether to run post_load methods..
         :return: A tuple of the form (`data`, `errors`)
         """
         many = self.many if many is None else bool(many)
-        partial = self.partial if partial is None else bool(partial)
+        if partial is None:
+            partial = self.partial
+        elif not utils.is_collection(partial):
```

Comment:
@justanr
yes, but how to deal with boolean values then, are you suggesting to completely change the semantics of partial ? My intention was to keep things backward compatible since this is a public api.
I agree with the type checking thing, I just need to know if we want to change partial completely, or keep around support for boolean values before I re-factor it.
Another option would be to add another parameter, e.g. `ignore_missing=tuple()` and leave partial untouched